iOS/Flows/Conversation/Cells/ConversationMessageCellLayout.swift

```diff
@@ -8,20 +8,26 @@
 
 import Foundation
 
+protocol ConversationMessageCellLayoutDelegate: AnyObject {
+    var message: Messageable? { get }
+}
+
 class ConversationMessageCellLayout: UICollectionViewFlowLayout {
 
+    unowned let messageDelegate: ConversationMessageCellLayoutDelegate
+
     override class var layoutAttributesClass: AnyClass {
         return ConversationMessageCellLayoutAttributes.self
     }
 
-    override init() {
+    init(messageDelegate: ConversationMessageCellLayoutDelegate) {
+        self.messageDelegate = messageDelegate
```

Comment:
As mentioned above, the layout class has a delegate to get the parent message of a thread.